### DISCUSSION TITLE
Is it possible for System.Numerics.Vector'n' structs to recieve cluster rounding member method

### DISCUSSION THREAD

**Discussion by Xyncgas** [dotnet/runtime#88963] [General]
Cluster rounding, for example if the difference between (x,y) is less than : (precision) then take the min(x,y) and returns a tuple of the same value otherwise return the original (x,y)

to perform this on vector would mean all the values inside vector3 for example would remain either unchanged or
if the difference between vec_x.X and vec_y.X is less than (precision) then now they have the same value, the result of the operation between two vectors should return two vectors

I wonder if it's SIMD applicable otherwise it might be non-sense to add this api to System.Numerics.Vector'n' structs

**Comment by tannergooding** [comment]
Could you provide a naive scalar implementation of what you're asking for? I'd like to confirm I'm properly understanding what's being asked for here.
